fix: LEAP-878: Fix shrinking textarea on edit

hlomzik opened the pull request:
We have TextArea's HtxTextBox (input when you edit text block) updating its height on edit. That was to auto-increase height of multiline blocks. But we stopped taking borders into account some time ago (https://github.com/HumanSignal/label-studio-frontend/commit/c8417d53463d87e4d54bf44d0508351973435a7e). Returning it back and fixing condition for empty scrollHeight fixes shrinking input.

+ fix position for "Delete Region" icon — it's in the middle now

<img width="486" alt="Screenshot 2024-04-03 at 16 02 11" src="https://github.com/HumanSignal/label-studio/assets/1607227/fe3fae2d-4006-413f-aa4e-5dd0991f68e8">


### PR fulfills these requirements
- [ ] Tests for the changes have been added/updated
- [ ] Docs have been added/updated
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance

### Describe the reason for change
Inputs and textareas used to edit `TextArea` tag text blocks were shrinking every time you type something into them.

### This change affects (describe how if yes)
- [ ] Performance
- [ ] Security
- [x] UX

### Does this PR introduce a breaking change?
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
- [ ] e2e (codecept)
- [ ] integration (cypress)
- [ ] unit (jest)

hlomzik committed:
fix: LEAP-878: Fix shrinking textarea on edit

We have TextArea's HtxTextBox (input when you edit text block) updating its height on edit.
That was to auto-increase height of multiline blocks. But we stopped taking borders into account
some time ago (https://github.com/HumanSignal/label-studio-frontend/commit/c8417d53463d87e4d54bf44d0508351973435a7e).
Returning it back and fixing condition for empty scrollHeight fixes shrinking input.

hlomzik committed:
Fix delete icon position

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8540883412

bmartel commented on web/libs/editor/src/components/HtxTextBox/HtxTextBox.js:
```diff
@@ -85,9 +85,13 @@ export class HtxTextBox extends React.Component {
   updateHeight = throttle(() => {
-    const height = this.inputRef.current?.scrollHeight || 0;
+    // very important to add borders to the height, otherwise input will be shrinking on every recalc
+    const borders = 2;
```
```suggestion
    const borders = 2;
```
We should compute the border contributed height in px from the element once, otherwise if there ever were to be a change that was made via css or inline style that affected the border-top and border-bottom width of the element, this would regress back to the original bug.

hlomzik replied:
now I set border-width in the code to always have it consistent

hlomzik committed:
Ensuring that borders are 1px exactly

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8572306178)
> Successfully merged:  76 files changed, 772 insertions(+), 1183 deletions(-)

hlomzik committed:
Merge branch 'develop' into 'fb-leap-878/textarea-height'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8572306178

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8572330410

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8621025372)
> Successfully merged:  14 files changed, 152 insertions(+), 51 deletions(-)

hlomzik committed:
Merge branch 'develop' into 'fb-leap-878/textarea-height'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8621025372

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8621088788
